### DISCUSSION TITLE
FAT seems wrong for mounted image

### DISCUSSION THREAD

**Discussion by jfrazier00** [joncampbell123/dosbox-x#5273] [General]
I'll try and explain this as accurately as I can in case I am doing something wrong/dumb.  I have an image file .IMG I have created from old 360 floppies using my Greaseweazle.  The conversion came through as 100%.  If I fire up the XT emulator over at PCjs https://www.pcjs.org/machines/pcx86/ibm/5160/cga/ and mount the .IMG file to a floppy drive, do a DIR and all looks great.

![image](https://github.com/user-attachments/assets/7088a337-63f8-462a-a44e-629b701baab1)

When I:
* Check the contents of an .ASM file (using TYPE), I get the expected assembly language source
* Check the contents of a .C file, it is my C source code
* Check the contents of .BAT file, it's a batch file
* Check contents of .BAS it is good ol' BASIC code
* Try to run EXE files, it's pretty darn close to what I would expect for some old circa 1983 CGA code

Now I try Dosbox-X and mount the same .IMG file using "Drive | B | Mount a disk or CD image file" and select the same IMG file, and do a DIR, I get the following.  Looks close and the file sizes look correct though the free space is a little off.

<img width="462" alt="image" src="https://github.com/user-attachments/assets/42783fb6-53f3-4683-bc54-23578ec0d06d">

 * Check a .BAT file and it looks like a binary
 * Check a .C file and it is assembly source
 * Check a .BAS file and it is C source
 * Check one of the .EXE and it is some C source
 * Try to run any EXE and they don't really work which makes sense given the above.

Am I totally doing something wrong here?

Also, I could log an issue, thought I would start here.

**Comment by rderooy** [comment]
Hmm... I'm wondering. a disk has two copies of the FAT, and perhaps one is corrupt, and MS-DOS and DOSBox-X are each using a different one by default?

Could you run chkdisk or scandisk against the disk image?

**Comment by jfrazier00** [comment]
Good thoughts, thanks.  Let's see what we have...

From DOSBox-X:
<img width="675" alt="image" src="https://github.com/user-attachments/assets/13037d50-5fc1-4c57-9d87-2ec242585842">

No scandisk as DOSBox-X is 5.00 by default and that was a 6.22 and up command.  The PCjs emulator is DOS 2.00 which is roughly period correct for an XT IIRC.

Here's the output from PCjs (I just answered no to the prompts for now):
![image](https://github.com/user-attachments/assets/f276e16b-f8da-4873-8c6f-a1393a2d8132)
![image](https://github.com/user-attachments/assets/dc22046e-0d60-4373-bda6-4786ce1eda48)
![image](https://github.com/user-attachments/assets/ec1efbd6-60cb-4e2e-851b-54ff2646dce0)

Realizing chkdsk is operating on volume a.  Been so long since I used on the regular.  Does the EXE need to be present on volume being checked?

**Comment by maron2000** [comment]
PC-DOS 1.0 comes with a 160kB floppy, boots fine on DOSBox-X so maybe some problems with your image.

**Comment by rderooy** [comment]
Another option I did not consider. Perhaps MS-DOS/PC DOS and DOSBox-X are parsing the same FAT by default, but the error handling of the corrupted table is different.
